### ISSUE TITLE
[a11y]: SR does not announce readonly status of form elements

### ISSUE THREAD

**Issue by dchyun** [carbon-design-system/carbon#22407]
### Package

@carbon/react, @carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

2.56.0

### React version

1.109.0

### Automated testing tool and ruleset

Manual

### Assistive technology

VoiceOver

### Description

For various form elements when the `readonly` attribute is present indicating a field is not editable this status is not properly indicated to a screen reader. The screen reader announces the same message regardless of if the field is `readonly`. This prevents a screen reader user from properly knowing the state of a form element. 
**Note:** In this issue macOS + VoiceOver was used for SR testing.

### Components with violations:
- Checkbox
- Dropdown
- Multi-select
- Number input
- Radio button
- Select
- Slider
- Toggle
- All fluid form components which are extensions of the ones listed above

### Example issue
The issue is similar across all components, but here is an example from the dropdown. In this example the dropdown is announced to the screen reader the same whether it is `readonly` or not.

**Default**

<img width="657" height="326" alt="Image" src="https://github.com/user-attachments/assets/0e29ade4-fe22-40c1-a8a2-d74e3eaf59f5" />

**Readonly**

<img width="661" height="267" alt="Image" src="https://github.com/user-attachments/assets/42afb571-4473-43a1-a014-5826c81eafc7" />

### Suggested solution
Instead of using a combination of various methods, most of which rely on `aria-readonly`, when a component is `readonly` add a `sr-only` message indicating this to the user.

#### aria-readonly
Various components rely on the usage of `aria-readonly="true"` to indicate read only status. The roles with witch this aria attribute can be applied are limited. Also, screen readers have a [history of poor support](https://adrianroselli.com/2022/11/brief-note-on-aria-readonly-support-html.html) for this and most do not announce anything. It is preferable to use the sr-only text solution outlined above instead.

#### Checkbox
The checkbox component uses a combination of setting `aria-disabled="true"` on the checkbox group, and `aria-readonly="true"` on each checkbox. This results in the checkbox group improperly being read as "disabled", when that is not accurate, as a SR user can still focus on the element. See above for the issues with using `aria-readonly`

#### Dropdown & Multi-select
The dropdown and multi-select use `aria-disabled="true"` when `readonly` is applied. This results in these components being improperly labeled as disabled when that is not accurate. 
**Note:** There are additional issues with the web component multi-select SR experience which will be captured in another issue. 

#### Number input
The number input uses the native `readonly` attribute on the `input`, which normally would be compliant. However the number input has `role="alert"`, and the `readonly` attribute is not supported for elements of that role.

<img width="258" height="61" alt="Image" src="https://github.com/user-attachments/assets/1f6f4a27-8f32-4053-81eb-d360c0249bf2" />

#### Radio button
The radio button has different issues across the web component and react implementation.

In the web component, `aria-readonly="true"` is used which does not accurately indicate readonly status.

In the react component, no `aria` attributes are used. There is no difference in the markup between the default and readonly states. 

#### Slider
The slider uses the native `readonly` attribute on an `input` of `type="number"` so there should not be any issues, but in testing no readonly status was being read. Not sure of the exact cause of this but it could be due to some issues with the aria connections between the labels and input.

In the web component, the `input` has no `id` and is not connected to any label. The `aria-label` of "Lower bound" is attached to the `cds` element not the `input` itself. 

Note: Not SR related but the web component also has an issue where the slider still has a hover effect when in readonly mode.

<img width="456" height="329" alt="Image" src="https://github.com/user-attachments/assets/4b5a7d86-5ca8-4513-b0ba-58844e683563" />

In the react component, the `for` attribute on the `label` is not connected to the `input`.

<img width="490" height="295" alt="Image" src="https://github.com/user-attachments/assets/06060b57-3142-4246-ad14-cbeca63ea939" />

#### Toggle
The toggle component does not appear to have any different markup between its `readonly` and default forms. The changes appear to be mostly stylistic. There are no `aria` arguments set or other changes.

### WCAG 2.1 Violation

Success Criterion 4.1.2 Name, Role, Value

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-checkbox--default&args=readonly:!true

### Steps to reproduce

1. Look at the public storybook for the web components or react library
2. For a given component set `readonly` to true in the actions panel
3. Open the page in isolation mode
4. Run VoiceOver on macOS or another screenreader
5. See if the screen reader announces anything related to read only status when the input is focused

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
